crates/nu-std/lib/help.nu

```diff
@@ -1,10 +1,124 @@
+#       retain consolidated for --concise
+# todo: `help command <foo>` gives error "no such command 'command'" -- bogus!
+# todo: but `help commands` should still give useful output.  Same for all kinds
+# todo: slow to format signature section!
+# todo: fix format where single flag or  double flag is not defined
+# todo: output a string for `help commands <specificCommand>`.  Today it's print output, may be why it's slow.
+
+# doc: --concise, not --verbose
+# doc: match item is <regex>.  use .* not *
+# doc: help -f <string> is unanchored match in multiple columns; 
+#      help <string> is whole word match, only in name column
+# doc: --find is a switch, not separate arg.  Can no longer say `help -f <foo> <bar>`; now that means search for pat "<foo> <bar"
+# doc: (Details section and Command Type
+# doc: show-modules: display imported commands with (*) prefix (maybe color later)
+# doc: only shows -h for custom cmds and for builtins that explicitly add it to signature
+# doc: operators info not searched for `help <operator>`, must use `help operators <operator>`
+# fixed: never use dark blue -- invisible on black screen
+# fixed: missing '{command,module,alias,operator,extrn}-not-found error; currently returns empty list but no error
+# fixed: extra blank lines between examples (commands)
+# fixed: now `help --find <thing>` that matches just 1 <thing> does display the item
+
+## themeing and formatting of various snippets
+
+def variable_shape_color [] {
+    ($env.config.color_config.shape_variable | default light_purple))
+}
+
+# <string> | colorize ...color => string
+# colorize input, not forgetting to reset afterwards.
+def colorize [...color_names: string] {
+    $"($color_names | each {|c| (ansi $c)} | str join '')($in)(ansi reset)"
+}
+
+def fmt-error [] {
+    ($in | colorize 'red')
+}
+
+def fmt-bold [] {
+    ($in | colorize 'attr_bold')
+}
+
+def fmt-header [] {
+    ($in | colorize green)
+}
+
+# format a placeholder token which is a data type: in <> and italicized
+def fmt-token [ ] {
+    ($"<($in)>" | fmt-type)
+}
+
+# format datatype name: italicized in blue
+def fmt-type [ ] {
+    ($in  | colorize attr_italic (variable_shape_color))
+}
+
+def pretty-cmd [] {
+    ($in | colorize default_dimmed default_italic)
+}
+
+## whole line formatting helpers
+
+# print section header and optional title line for the section
+def print-help-header [
+    text: string
+    title?: string = "" # optional title
+    --no_newline (-n)   # suppress the default leading NL
+] {
+    if not $no_newline {
+        print ""
+    }
+    print $"($text | fmt-header) ($title)"
+}
+
+# print detail line under section (indented)
+def print-help-detail [text: string, indent?:int = 2] {
+    $text | lines | each {|l| print $"(' ' * $indent)($l)" }
+}
+
+# Print section header and content in one fell swoop.  Or skip it if empty.
+def print-help-section [section:string,  # section name, skip printing if ''
+                        body?:string = ""
```

Comment:
Nit:
```suggestion
def print-help-section [
  section: string, # section name, skip printing if ''
  body?: string = ""
```

Reply:
Agree the formatting is not standard.  Wish there were a nufmt tool that would handle this automatically.

Reply:
I believe @amtoine is working on it 😉